crates/sui-core/src/authority_server.rs

```diff
@@ -325,15 +290,64 @@ impl Validator for ValidatorService {
             tx_kind = certificate.data.kind_as_str()
         );
 
-        let response = self
-            .state
+        let response = state
             .handle_certificate(certificate)
             .instrument(span)
             .await
             .map_err(|e| tonic::Status::internal(e.to_string()))?;
 
         Ok(tonic::Response::new(response))
     }
+}
+
+#[async_trait]
+impl Validator for ValidatorService {
+    async fn transaction(
+        &self,
+        request: tonic::Request<Transaction>,
+    ) -> Result<tonic::Response<TransactionInfoResponse>, tonic::Status> {
+        let mut transaction = request.into_inner();
+
+        transaction
+            .verify()
+            .map_err(|e| tonic::Status::invalid_argument(e.to_string()))?;
+        //TODO This is really really bad, we should have different types for signature-verified transactions
+        transaction.is_verified = true;
+
+        let tx_digest = transaction.digest();
```

Comment:
Should we stick all the below in its own task too? Just to make sure that signing also does not get interrupted?

Reply:
yeah sure, let me edit the pr